kubernetes.go

```diff
@@ -320,13 +330,13 @@ func (c *KubernetesClient) Watch() error {
 			c.eventForwarder(e)
 		}
 
-		if !more {
+		if !more || e.Type == watch.Error {
```

Comment:
If a watch error is received, is the connection with the API still up?
If that is the case, should we disconnect before trying to connect again to the API ? is it managed by connect method ?